helm/helm.go

```diff
@@ -176,22 +177,32 @@ func GetChartFile(file []byte, fileName string) (string, error) {
 }
 
 //DeleteAllDeployment deletes all Helm deployment
-func DeleteAllDeployment(kubeconfig []byte) error {
-	log.Info("Getting deployments....")
+func DeleteAllDeployment(log logrus.FieldLogger, kubeconfig []byte) error {
+	log.Info("getting deployments....")
 	filter := ""
 	releaseResp, err := ListDeployments(&filter, "", kubeconfig)
 	if err != nil {
-		return err
+		return emperror.Wrap(err, "failed to get deployments")
 	}
-	log.Info("Starting deleting deployments")
+	log.Info("start deleting deployments")
 	if releaseResp != nil {
+		// the returned release items are unique by release name and status
+		// e.g. release name = release1, status = PENDING_UPGRADE
+		//      release name = release1, status = DEPLOYED
+		//
+		// we need only the release name for deleting a release
+		releases := make(map[string]bool)
 		for _, r := range releaseResp.Releases {
-			log.Info("Trying to delete deployment ", r.Name)
-			err := DeleteDeployment(r.Name, kubeconfig)
+			releases[r.Name] = true
+		}
+
+		for release := range releases {
```

Comment:
Why not merge the two loops and skip already deleted releases?

Reply:
done